### ISSUE TITLE
Generate GraphQL schema for local service variable declaration

### ISSUE THREAD

**Issue by MohamedSabthar** [ballerina-platform/ballerina-library#3317]
**Description:**
The current implementation of graphql module generate schema for both service declaration and module level (graphql:Service) variable declaration. But it not yet generates graphql schema for local variable declaration.

ex: bellow code doesn't generate a schema
```ballerina
import ballerina/lang.runtime;
import ballerina/graphql;

public function main() returns error? {
    // local service variable declaration
    graphql:Service backendService = service object {
        resource function get greeting() returns string {
            return "Hello, World";
        }
    };
    
    graphql:Listener serverEP = check new(4000);
    check serverEP.attach(backendService, "graphql");
    check serverEP.'start();
    runtime:registerListener(serverEP);
}
```

Ideally graphql module should generate schema for local variable declaration in any block scope (function definition, if else body, object field, and so on)

**Describe your problem(s)**

**Describe your solution(s)**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by MohamedSabthar** [comment]
The graphql schema is generated in compile time and attached as a field of @graphql:ServiceConfig annotation of the graphql service object/declaration.

This is done by the graphql compiler plugin by following two steps
Step 1, the service object Nodes are collected in Map (by a SyntaxNodeAnaylysisTask)
In step 2, the collected Nodes are modified (by a SourceModiferTask) 
Finally, The generated schema is accessed in runtime from the annotation to run the service.

In the Second task, the node modification is performed by traversing the ModulePartNode (of the document) in a top-down manner. 
If the service node collected in the Map (from step 1) is found during the traversal then the node is modified with the annotation and the parent nodes are recursively modified until it reaches the ModulePartNode.

The schema generation for module-level service variable declaration and service declaration already implemented using the mentioned approach. But it’s not efficient to implement the local service variable declaration as we need to consider a lot of local scopes (function definition, if else body, object field, etc...).
A simpler way to replace the node from the tree is required.

@MaryamZi @hasithaa @warunalakshitha is there a simpler compiler plugin API to achieve this ?

**Comment by MohamedSabthar** [comment]
+ @shafreenAnfar

**Comment by hasithaa** [comment]
@MohamedSabthar Could you point me to the implementation of SyntaxNodeAnaylysisTask and SourceModiferTask. 

One way to achieve this is you could listen to object contractor events and add those to the map. Then later annotate it with  graphql:ServiceConfig using SourceModiferTask.

**Comment by MohamedSabthar** [comment]
> @MohamedSabthar Could you point me to the implementation of SyntaxNodeAnaylysisTask and SourceModiferTask.
@hasithaa these are the implementation [SyntaxNodeAnaylysisTask](https://github.com/ballerina-platform/module-ballerina-graphql/blob/master/compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/ServiceAnalysisTask.java#L41), [SourceModifierTask](https://github.com/ballerina-platform/module-ballerina-graphql/blob/master/compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/schema/generator/GraphqlSourceModifier.java#L84)

**Comment by MohamedSabthar** [comment]
Had a chat with @hasithaa @azinneera @ThisaruGuruge and @DimuthuMadushan on this issue.
A new feature is requested [#38054](https://github.com/ballerina-platform/ballerina-lang/issues/38054)

**Comment by MohamedSabthar** [comment]
Blocked on [#38054](https://github.com/ballerina-platform/ballerina-lang/issues/38054)

**Comment by hasithaa** [comment]
@MohamedSabthar Shall we try to implement this with current design as we discussed during the call. Main problem with current design is not scalable when we have multiple code modifiers/modifications.

**Comment by MohamedSabthar** [comment]
Since there is low ROI, involves complex logic, and there's no urgency of this feature, the implementation of this functionality put on hold
